public/js/components/Expressions.js

```diff
@@ -3,70 +3,134 @@ const { connect } = require("react-redux");
 const { bindActionCreators } = require("redux");
 const ImPropTypes = require("react-immutable-proptypes");
 // const classnames = require("classnames");
+const Svg = require("./utils/Svg");
 const actions = require("../actions");
 const { getExpressions, getPause } = require("../selectors");
-// const ObjectInspector = React.createFactory(require("./ObjectInspector"));
+const Rep = React.createFactory(require("./Rep"));
 // const { truncateStr } = require("../utils/utils");
 const { DOM: dom, PropTypes } = React;
 
 require("./Expressions.css");
 
 const Expressions = React.createClass({
   propTypes: {
-    pauseInfo: ImPropTypes.map,
     expressions: ImPropTypes.list,
     addExpression: PropTypes.func,
     updateExpression: PropTypes.func,
-    evaluateExpression: PropTypes.func,
-    loadObjectProperties: PropTypes.func,
-    command: PropTypes.func
+    deleteExpression: PropTypes.func
   },
 
   displayName: "Expressions",
 
-  addExpression(e) {
+  addExpression({ id }, e) {
     if (e.key === "Enter") {
+      const { addExpression } = this.props;
```

Comment:
nit: it can be nicer to use a guard statement. it avoids the extra nesting and implies your looking to limit the input to the function early on.

https://en.wikipedia.org/wiki/Guard_(computer_science)

``if (e.key !== "Enter") { 
  return
}
```